x-pack/docs/en/security/authorization/run-as-privilege.asciidoc

```diff
@@ -20,16 +20,120 @@ permission. For example, the following role grants permission to submit request
 on behalf of `jacknich` or `redeniro`:
 
 [source,js]
----------------------------------------------------
+----
 {
   "run_as" : [ "jacknich", "rdeniro" ]
 }
----------------------------------------------------
+----
 
 To submit a request as another user, you specify the user in the
 `es-security-runas-user` request header. For example:
 
 [source,shell]
----------------------------------------------------
+----
 curl -H "es-security-runas-user: jacknich"  -u es_admin -XGET 'http://localhost:9200/'
----------------------------------------------------
+----
+
+[[run-as-privilege-apply]]
+== Apply the `run_as` privilege to roles
+
+You can apply the `run_as` privilege when creating roles with the
+<<security-api-put-role,create or update roles API>>. Users who are assigned
+a role that contains the `run_as` privilege inherit all privileges from their
+role, and can also submit requests on behalf of the indicated users.
+
+When <<authorization_realms,delegated authorization>> is disabled, {es} handles
```

Comment:
This might be too specific to JWT realm. Sorry I did not make it clear for the general case.

In general, realms can support a broad list of options for role lookup.
1. Realm can do direct lookup of a user with predefined roles. Two sub-options are API or local file. Example is native realm users with predefined roles. Another is LDAP user lookup in an LDAP server. Users in a file is deprecated.
2. Realm can do indirect lookup via role mapping rules. Two sub-options are API or local file. Example is JWT or OIDC realms. Role mapping in a file is deprecated.
3. Realm can do delegated authorization (ex: JWT realm), which overrides options 1 & 2.

JWT realm supports 2 & 3. For option 2, the file sub-option was not implemented. The file option is only in a couple of realms.

Reply:
@justincr-elastic, let me know if [these changes](https://github.com/elastic/elasticsearch/pull/86411/commits/03cdb74c908274a032a46ca3f9a1f34ccb4e03ae#diff-2aaf178d73325e46e4c68d44253ddd180e189bc3540304e2c0b0468d90a15ee2R46-R55) encapsulate what you are describing.